js/test/input-mask.test.js

```diff
@@ -0,0 +1,199 @@
+    await page.goto(`${config.BASE_URL}/input-mask.php`, {
+      waitUntil: "domcontentloaded",
+    });
+
+    const telInput = "#tel";
+    await page.focus(telInput);
+    await page.keyboard.type("1234567890");
+    await page.keyboard.press("Backspace");
+    await page.keyboard.press("ArrowLeft");
+    await page.keyboard.type("1");
+    const maskedTelephone = await page.$eval(
+      "span.enable-input-mask__mask-pre-val",
+      (span) => span.textContent
+    );
+    expect(maskedTelephone).toBe("123-456-781");
+  });
+
+  // Test mouse interaction
+  it("should correctly render visual cursor and the input cursor on mouse Interactions", async () => {
+    const page = await desktopBrowser.newPage();
+    await page.goto(`${config.BASE_URL}/input-mask.php`, {
+      waitUntil: "domcontentloaded",
+    });
+
+    const telephoneInput = "#tel";
+
+    // bounding box of input
+    const inputBoundingBox = await page.$eval(telephoneInput, (input) => {
+      const { x, y, width, height } = input.getBoundingClientRect();
+      return { x, y, width, height };
+    });
+
+    // Calculate the middle of the span
+    const middleX = inputBoundingBox.x + inputBoundingBox.width / 2;
+    const middleY = inputBoundingBox.y + inputBoundingBox.height / 2;
+
+    //Initial cursor position, before mouse Interaction
+    const initialCursorPosition = await captureCursorPosition();
+
+    //explicitly move the mouse and click
+    await page.mouse.move(middleX, middleY);
+    await page.mouse.click(middleX, middleY);
+
+    // get the cursor position ofter click, the assumption is after mouse move, an onclick the cursor doesn't Move
+    const cursorPosition = await captureCursorPosition();
+
+    expect(cursorPosition.x).toBeCloseTo(initialCursorPosition.x, 2);
+    expect(cursorPosition.y).toBeCloseTo(initialCursorPosition.y, 2);
+  });
+
+  // Mouse Selection
+  it("should select characters in the Tel Input field - Mouse", async () => {
+    const page = await desktopBrowser.newPage();
+    await installMouseHelper(page);
+    await page.goto(`${config.BASE_URL}/input-mask.php`, {
+      waitUntil: "domcontentloaded",
+    });
+    const telephoneInput = "#tel";
+    const expectedSelection = "12345";
+    await page.type(telephoneInput, "1234567890");
+    // bounding box of input
+    const inputBoundingBox = await page.$eval(telephoneInput, (input) => {
+      const { x, y, width, height } = input.getBoundingClientRect();
+      return { x, y, width, height };
+    });
+    //Click the input and move to the start of the field
+    await page.click(telephoneInput);
+    await page.mouse.move(inputBoundingBox.x, inputBoundingBox.y);
+    // Select using
+    await page.mouse.down();
+    await page.mouse.move(inputBoundingBox.x + 50, inputBoundingBox.y);
+    await page.mouse.up();
+    //get the selection
+    const selectedText = await page.$eval(
+      "span.enable-input-mask__mask-pre-val",
+      (span) => {
```

Comment:
_nit_: Since 'span' isn't used, we shouldn't define it
```suggestion
      () => {
```